File: src/main/java/org/tarantool/queue/TaskStatus.java
package org.tarantool.queue;

public enum TaskStatus {
    READY("r"),
    TAKEN("t"),
    EXECUTED("-"),
    BURIED("!"),
    DELAYED("~");

    private String symbol;

    TaskStatus(String symbol) {
        this.symbol = symbol;
    }

    public static TaskStatus getBySymbol(String symbol) {
        switch (symbol) {
            case "r":
                return TaskStatus.READY;
            case "t":
                return TaskStatus.TAKEN;
            case "-":
                return TaskStatus.EXECUTED;
            case "!":
                return TaskStatus.BURIED;
            case "~":
                return TaskStatus.DELAYED;
            default:
                throw new IllegalArgumentException("Unknown symbol of task status: " + symbol);
        }
    }
}

File: src/main/java/org/tarantool/queue/TaskStatus.kt
package org.tarantool.queue

/**
 * Created by GrIfOn on 03.02.2018.
 */
enum class TaskStatus(val value: String) {
    READY("r"), TAKEN("t"), EXECUTED("-"), BURIED("!"), DELAYED("~");

    companion object {
        fun getByValue(value: String): TaskStatus? {
            return when(value) {
                "r" -> TaskStatus.READY
                "t" -> TaskStatus.TAKEN
                "-" -> TaskStatus.EXECUTED
                "!" -> TaskStatus.BURIED
                "~" -> TaskStatus.DELAYED
                else -> null
            }
        }
    }
}